modular_skyrat/modules/xenoarch/code/modules/research/xenoarch/xenoarch_machine.dm

```diff
@@ -1,138 +1,164 @@
 
-/obj/machinery/xenoarch/proc/insert_xeno_item(obj/item/insert_item, mob/living/user)
-	insert_item.forceMove(holding_storage)
-	playsound(src, 'sound/machines/click.ogg', 50, TRUE)
-	to_chat(user, span_notice("The [insert_item] has been inserted into [src]."))
-	world_compare = world.time + process_speed
-	addtimer(CALLBACK(src, PROC_REF(do_machine_process)), process_speed)
+/obj/machinery/xenoarch/wrench_act(mob/living/user, obj/item/tool)
+	. = ..()
+
+	if(default_unfasten_wrench(user, tool))
+		return ITEM_INTERACT_SUCCESS
 
-/obj/machinery/xenoarch/attack_hand(mob/living/user, list/modifiers)
+/obj/machinery/xenoarch/screwdriver_act(mob/living/user, obj/item/tool)
 	. = ..()
-	var/src_turf = get_turf(src)
-	var/user_choice = tgui_alert(user, "Are you sure you want to remove all the items?", "Remove Items?", list("Yes", "No"))
-	if(user_choice != "Yes")
-		return
-	for(var/move_out_item in holding_storage.contents)
-		if(!isobj(move_out_item))
-			continue
-		var/obj/move_out_obj = move_out_item
-		move_out_obj.forceMove(src_turf)
 
-/obj/machinery/xenoarch/examine(mob/user)
+	toggle_panel_open()
+	to_chat(user, span_notice("You [panel_open ? "open":"close"] the maintenance panel of [src]."))
+	tool.play_tool_sound(src)
+	return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/xenoarch/crowbar_act(mob/living/user, obj/item/tool)
 	. = ..()
-	if(holding_storage?.contents.len)
-		var/is_singular = holding_storage.contents.len == 1 ? TRUE : FALSE
-		. += span_notice("There [is_singular ? "is" : "are"] [holding_storage.contents.len] [is_singular ? "item" : "items"] currently in the buffer.")
-		. += span_warning("You can remove the contents, but the machine is currently trying to process the items!")
+
+	if(default_deconstruction_crowbar(tool))
+		return ITEM_INTERACT_SUCCESS
 
 /obj/machinery/xenoarch/researcher
 	name = "xenoarch researcher"
 	desc = "A machine that is used to condense strange rocks, useless relics, and broken objects into bigger artifacts."
 	icon_state = "researcher"
 	circuit = /obj/item/circuitboard/machine/xenoarch_machine/xenoarch_researcher
-	///A variable that goes from 0 to 150. Depending on what is processed, increases the value. Once at or above 150, spawns an anomalous crystal.
+	/// the amount of research that is currently done
 	var/current_research = 0
+	/// the max amount of value we can have
+	var/max_research = 300
+	/// the value of each accepted item
+	var/list/accepted_types = list(
+		/obj/item/xenoarch/strange_rock = 1,
+		/obj/item/xenoarch/useless_relic = 5,
+		/obj/item/xenoarch/useless_relic/magnified = 10,
+		/obj/item/xenoarch/broken_item = 10,
+	)
 
 /obj/machinery/xenoarch/researcher/examine(mob/user)
 	. = ..()
-	. += span_notice("[current_research]/150 research points. Research more xenoarchaeological items.")
+
+	. += span_notice("<br>[current_research]/[max_research] research available.")
+	. += span_notice("L-Click to insert items or take out all the strange rocks. R-Click to use research points.")
```

Comment:
Could turn these into tooltips at the top of the screen?

Reply:
I... probably could, yes.